ce/api/geo.py

```diff
@@ -124,5 +125,14 @@ def polygonToMask(nc, fname, poly, variable):
             raise Exception("Unable to determine projection parameters for GDAL "
                             "dataset {}".format(dst_name))
 
-        mask = rasterize((poly,), out_shape=raster.shape, transform=raster.affine, all_touched=True)
-    return mask == 0
+        the_array, _ = rio_mask(raster, [mapping(poly)], crop=False, all_touched=True)
+
+    # Weirdly rasterio's mask operation sets, but doesn't respect the
+    # Fill_Value, scale_factor, or add_offset
+    var = nc.variables[variable]
+    the_array.mask = the_array==the_array.fill_value
+    scale_factor = var.scale_factor if hasattr(var, 'scale_factor') else 1.0
+    add_offset = var.add_offset if hasattr(var, 'add_offset') else 0.0
+
+    the_array = the_array * scale_factor + add_offset
+    return the_array
```

Comment:
where the word `mask` is interpreted as a verb -- so perhaps a different word or wording would be superior